tests/aio_test.c

```diff
@@ -0,0 +1,190 @@
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strings.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+#include <linux/aio_abi.h>
+
+/*
+ * A simple test to exercise the io_*() system calls.
+ * Usage:
+ *   aio_test number_of_files
+ */
+
+// We only support up to 222 concurrent io's in this test.
+// Arbitrarily chosen.
+#define MAX_FDS 222
+// We write CHUNKSIZE bytes into each file for this test.
+#define CHUNKSIZE 4096
+char* progname = "noname";
+void usage(void)
+{
+   fprintf(stderr, "Usage: %s number_of_files\n", progname);
+}
+
+int main(int argc, char* argv[])
+{
+   int nf;
+   long rc = 0;
+   int chunksize = CHUNKSIZE;
+   char buffer[MAX_FDS * CHUNKSIZE];
+   int fdlist[MAX_FDS];
+   struct iocb iocb[MAX_FDS];
+   struct iocb* iocblist[MAX_FDS];
+   struct io_event eventlist[MAX_FDS];
+   aio_context_t iocontext;
+   char filename[128];
+
+   progname = argv[0];
+
+   if (argc != 2) {
+      usage();
+      return 1;
+   }
+   nf = atoi(argv[1]);
+   if (nf <= 0) {
+      usage();
+      return 1;
+   }
+   fprintf(stdout, "Using %d files for this test run\n", nf);
+
+   // create data for test file contents and open test files
+   // and setup async io control blocks.
+   for (int i = 0; i < nf; i++) {
+      // fill buffer
+      memset(&buffer[i * chunksize], i, chunksize);
+
+      // open files, pid may not be unique in a container
+      snprintf(filename, sizeof(filename), "/tmp/aio_file_%d_%d", getpid(), i);
```

Comment:
I don't think getpid() is good enoughto generate a unique name in a container.
Wish there was a uuid generator function in the c library.